### PR TITLE
HSEARCH-4067 Remove "(default)"/"(no default)" from configuration examples in the documentation

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4067

### CODE REVIEW

File: documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc [hunk 1]
```diff
@@ -239,31 +241,34 @@ Timeouts::
 [source]
 ----
 # hibernate.search.backend.request_timeout = (not defined by default)
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc:243]: I think you should replace this one with a real example, too.

File: documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc [hunk 2]
```diff
@@ -132,14 +134,14 @@ Automatic discovery is controlled by the following properties:
...
+hibernate.search.backend.discovery.enabled = false
+hibernate.search.backend.discovery.refresh_interval = 10
 ----
 
 * `discovery.enabled` defines whether the feature is enabled.
-Expects a boolean value.
+Expects a boolean value. The default for this property is `false`.
 * `discovery.refresh_interval` defines the interval between two executions of the automatic discovery.
-Expects a positive integer, in seconds.
+Expects a positive integer, in seconds. The default for this property is `10`.
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc:144]: You forgot to update authentication properties (below).

File: documentation/src/main/asciidoc/reference/backend-lucene.asciidoc [hunk 3]
```diff
@@ -238,11 +239,12 @@ Multiple strategies are available:
...
 # To configure the defaults for all indexes:
 hibernate.search.backend.sharding.strategy = hash
-hibernate.search.backend.sharding.number_of_shards = 2 (no default)
+hibernate.search.backend.sharding.number_of_shards = 2
 # To configure a specific index:
 hibernate.search.backend.indexes.<index name>.sharding.strategy = hash
-hibernate.search.backend.indexes.<index name>.sharding.number_of_shards = 2 (no default)
+hibernate.search.backend.indexes.<index name>.sharding.number_of_shards = 2
 ----
+There is no default for this property.
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/reference/backend-lucene.asciidoc:247]: I wouldn't include this line: it's confusing, as there are multiple properties listed above.
Also, we already say that `number_of_shards` is required just one line below.

Just remove the `(no default)`.

File: documentation/src/main/asciidoc/reference/backend-lucene.asciidoc [hunk 4]
```diff
@@ -264,11 +266,12 @@ to be brought down to a smaller number (e.g. "all integers").
...
 # To configure the defaults for all indexes:
 hibernate.search.backend.sharding.strategy = explicit
-hibernate.search.backend.sharding.shard_identifiers = fr,en,de (no default)
+hibernate.search.backend.sharding.shard_identifiers = fr,en,de
 # To configure a specific index:
 hibernate.search.backend.indexes.<index name>.sharding.strategy = explicit
-hibernate.search.backend.indexes.<index name>.sharding.shard_identifiers = fr,en,de (no default)
+hibernate.search.backend.indexes.<index name>.sharding.shard_identifiers = fr,en,de
 ----
+There is no default for this property.
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/reference/backend-lucene.asciidoc:274]: Same as above: this is confusing, let's remove this line?

**Review by yrodiere — CHANGES_REQUESTED** [review]

**Comment by fax4ever** [comment]
Thanks for the review. Applied all the comments and force pushed.

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**